Fix: MonkeyTest UI tests flaky in iOS CI (background assertion timeout)

MonkeyTest UI tests are intermittently failing in the iOS CI workflow (`ios.yml`) on the `macos-15` runner with Xcode 16.4.

## Failing tests

- `MonkeyTest.testAllScreensNavigation()` -- times out during screen navigation
- `MonkeyTest.testBackgroundForegroundWithAudio()` -- fails with "Timed out while acquiring background assertion"

## Error details

```
MonkeyTest.swift:359: error: -[UkuleleCompanionUITests.MonkeyTest testBackgroundForegroundWithAudio] :
Failed to get background assertion for target app with pid 28212:
Timed out while acquiring background assertion.
```

## Context

- A previous fix (`7b00176`) addressed the same background assertion timeout in the stress test workflow (`ios-stress.yml`), but the issue persists in the main iOS CI workflow.
- The other MonkeyTest cases (`testMonkeyRandom`, `testRapidRotation`, `testTunerRapidStartStop`) are correctly skipped in CI when `RUN_STRESS_TESTS` is not set.
- `testAllScreensNavigation` and `testBackgroundForegroundWithAudio` are not guarded by `RUN_STRESS_TESTS` and run in every CI build, causing flaky failures.

## Possible fix

These two tests appear to be stress/UI tests that should either:
1. Be guarded by the `RUN_STRESS_TESTS` environment variable like the other MonkeyTest cases, or
2. Be moved to the dedicated `ios-stress.yml` workflow where simulator flakiness is expected and doesn't block PRs.

## Recent occurrence

- PR #68 (Codecov integration) -- unrelated changes, iOS CI failed due to this flakiness.

--- a/iosApp/UkuleleCompanionUITests/MonkeyTest.swift
+++ b/iosApp/UkuleleCompanionUITests/MonkeyTest.swift
@@ -58,7 +58,7 @@ final class MonkeyTest: XCTestCase {
 
     override func tearDownWithError() throws {
         XCUIDevice.shared.orientation = .portrait
-        app.terminate()
+        app?.terminate()
     }
 
     // MARK: - Main Monkey Test

